packages/query-core/src/queryClient.ts

```diff
@@ -81,6 +83,9 @@ export class QueryClient {
   }
 
   mount(): void {
+    this.mountCount++
+    if (this.mountCount !== 1) return
+
```

Comment:
could we alternatively just call `this.unmount()` at the beginning of the `mount` function here? Basically saying: "If you `mount` a second time, we `unmount` the first instance", as opposed to: "If you `mount` a second time, we don't do anything because you're already mounted" ....

I've tried your test with this approach and it passed.

Reply:
Thanks for taking a look at this! You're right, the test perhaps could be better, but that approach would break if we mounted two roots that share a client, then unmounted just one. In that circumstance, we'd have unsubscribed even though there is still one root mounted, hence the need for a count. 😀

Reply:
@TkDodo I just pushed up an additional test to cover the case I described above.